src/components/datePicker/DatePicker.tsx

```diff
@@ -52,22 +51,21 @@ export class DatePicker extends React.Component<IDatePickerProps, any> {
   private setToToday() {
     const date = new Date();
     this.dateInput.value = this.getStringFromDate(date);
-    this.handleChange();
+    this.handleChangeDate();
   }
 
-  private handleChange() {
+  private handleChangeDate() {
     const date: Date = this.getDateFromString(this.dateInput.value);
 
     if (date.getDate()) {
-      this.props.onChange(date, this.props.upperLimit);
+      this.props.onBlur(date, this.props.upperLimit);
```

Comment:
suppose I change the value in the input, and than click on "Apply" without clicking elsewhere, my click won't work, only the second one and up will work. Could a debounced function work here?